app/importers/clients/sftp_client.rb

```diff
@@ -39,20 +41,52 @@ def dir_entries(*args)
   # This returns an object that will reveal secure data if printed
   def sftp_session
     raise "SFTP information missing" unless sftp_info_present?
+
     if @sftp_session.nil?
-      @sftp_session = Net::SFTP.start(host, user, auth_mechanism)
+      start_sftp_session
```

Comment:
Does this need to assign `@sftp_session` too?  Or could we move the assignment here rather than doing it in each method?